### PR TITLE
History functionality

### PR DESCRIPTION
**Pull request by omkarpatke**
# Added History Functionality

**Screenshots: **
hinstory-page

**Files To check:**
- History.css
- History.js
- historyContext.js
- HomePage.js
- apiCalls.js

### CODE REVIEW

File: src/Pages/HomePage/HomePage.css [hunk 1]
```diff
@@ -0,0 +1,56 @@
+:root{
+    --bg-color: #fcfcfc;
+    --category-color:#e7e2e2;
```
Comment (nikhiljugale007) [inline src/Pages/HomePage/HomePage.css]: use color variables instead of direct values

File: src/Pages/HomePage/HomePage.css [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    padding: 1rem;
+}
+
+.video{
+   background-color: var(--white-color);
+   max-width: 300px;
+   max-height: auto;
+   gap: 1rem;
+   cursor: pointer;
+   height: 320px;
```
Comment (nikhiljugale007) [inline src/Pages/HomePage/HomePage.css]: try to use rems instead of px

File: src/Pages/History/History.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import './History.css';
+import { SideBar , CategoryBar } from '../../components/index';
+import { Link } from 'react-router-dom';
+import { removeHistoryVideo , removeAllHistory  } from '../../api-calls/api-calls';
+import { useHistoryContext } from '../../context/HistoryContext';
+
+export function History() {
+    const { historyVideosState , historyVideosDispatch } = useHistoryContext();
+    let historyVideos;
```
Comment (nikhiljugale007) [inline src/Pages/History/History.js:10]: you should use useState for historyVideos instead of  let variable.

**Review by nikhiljugale007 — COMMENTED** [review]
The history functionality is not working for me. Can you check it once.
Added some suggestion in code.

**Comment by omkarpatke** [comment]
> The history functionality is not working for me. Can you check it once. Added some suggestion in code.

I Checked this it's working on my machine.

File: src/Pages/HomePage/HomePage.css (hunk 1 above)
Comment (omkarpatke) [reply src/Pages/HomePage/HomePage.css]: but this is the root.

File: src/Pages/History/History.js (hunk 3 above)
Comment (omkarpatke) [reply src/Pages/History/History.js:10]: ok, I will work on that.

File: src/Pages/HomePage/HomePage.css (hunk 2 above)
Comment (omkarpatke) [reply src/Pages/HomePage/HomePage.css]: thanks for the suggestion, changing px to rem.

**Review by ravikumar1002 — COMMENTED** [review]
UI looks good 
clean code , easy to understand

**Comment by ravikumar1002** [comment]
> UI looks good clean code , easy to understand

 a problem in history that I was facing
*  I logged in with dummy data
* after that I open a video
* But when I went to the page of history, that video was not showing
* When I refreshed the site on history page it was showing broken link
* and then I navigate to the home page via the link edit
After that I opened the video and then it was showing in the video history page

 I  logged out and refreshed the page,  then upper mentioned problems was repeating again

**Comment by ravikumar1002** [comment]
> > UI looks good clean code , easy to understand
> 
> a problem in history that I was facing
> 
> * I logged in with dummy data
> * after that I open a video
> * But when I went to the page of history, that video was not showing
> * When I refreshed the site on history page it was showing broken link
> * and then I navigate to the home page via the link edit
>   After that I opened the video and then it was showing in the video history page
> 
> I logged out and refreshed the page, then upper mentioned problems was repeating again

you can try this with preview link in Incognito tab

**Comment by omkarpatke** [comment]
> > > UI looks good clean code , easy to understand
> > 
> > 
> > a problem in history that I was facing
> > 
> > * I logged in with dummy data
> > * after that I open a video
> > * But when I went to the page of history, that video was not showing
> > * When I refreshed the site on history page it was showing broken link
> > * and then I navigate to the home page via the link edit
> >   After that I opened the video and then it was showing in the video history page
> > 
> > I logged out and refreshed the page, then upper mentioned problems was repeating again
> 
> you can try this with preview link in Incognito tab

Hey Ravi, First thanks for your response. I tried with incognito tab also but it works.

**Comment by ravikumar1002** [comment]
> > > > UI looks good clean code , easy to understand
> > > 
> > > 
> > > a problem in history that I was facing
> > > 
> > > * I logged in with dummy data
> > > * after that I open a video
> > > * But when I went to the page of history, that video was not showing
> > > * When I refreshed the site on history page it was showing broken link
> > > * and then I navigate to the home page via the link edit
> > >   After that I opened the video and then it was showing in the video history page
> > > 
> > > I logged out and refreshed the page, then upper mentioned problems was repeating again
> > 
> > 
> > you can try this with preview link in Incognito tab
> 
> Hey Ravi, First thanks for your response. I tried with incognito tab also but it works.

This is what I am trying to say
https://user-images.githubusercontent.com/86847625/168964349-f5819fb2-1343-463a-ae86-05ccefcf175c.mp4
if this is working fine in your machine that's fine

**Review by ravikumar1002 — APPROVED** [review]

**Pull request merged**